### PR TITLE
Add sort option: Only this folder

### PR DESCRIPTION
**Pull request by ningshu**
Fixes [#528](https://github.com/TeamAmaze/AmazeFileManager/issues/528)

### CODE REVIEW

File: app/src/main/java/com/amaze/filemanager/ui/dialogs/GeneralDialogCreation.java [hunk 1]
```diff
@@ -922,32 +926,61 @@ public static void showCompressDialog(final MainActivity m, final ArrayList<Hybr
...
+        final String path = m.getCurrentPath();
         int accentColor = m.getMainActivity().getAccent();
         String[] sort = m.getResources().getStringArray(R.array.sortby);
-        int current = Integer.parseInt(sharedPref.getString("sortby", "0"));
+        int current = SortHandler.getSortType(m.getContext(), path);
         MaterialDialog.Builder a = new MaterialDialog.Builder(m.getActivity());
         a.theme(appTheme.getMaterialDialogTheme());
         a.items(sort).itemsCallbackSingleChoice(current > 3 ? current - 4 : current,
                 (dialog, view, which, text) -> true);
+        final Set<String> onlyThisFloders = sharedPref.getStringSet(PREFERENCE_SORTBY_ONLY_THIS, new HashSet<>());
```
Comment (EmmanuelMess) [inline app/src/main/java/com/amaze/filemanager/ui/dialogs/GeneralDialogCreation.java]: > <p>Note that you <em>must not</em> modify the set instance returned
> by this call.  The consistency of the stored data is not guaranteed
> if you do, nor is your ability to modify the instance at all.

-- [source](https://developer.android.com/reference/android/content/SharedPreferences.html#getStringSet(java.lang.String,%20java.util.Set%3Cjava.lang.String%3E))

I like your use of Set for efficiently finding if Strings exist.

**Comment by ningshu** [comment]
Error usage on api [getStringSet](https://developer.android.com/reference/android/content/SharedPreferences.html#getStringSet(java.lang.String,%20java.util.Set%3Cjava.lang.String%3E))  has been fixed.

**Comment by VishalNehra** [comment]
@TranceLove please test this PR and give your views.
I tested on my Nexus 6p it's never successful no matter what I do. It always fails with logs saying the table doesn't exist. I know it's not possible since all queries written are correct. This might be my device (ROM) specific. 
I also tested on N5x emulator running P with success.

**Comment by TranceLove** [comment]
A possible cure to the symptom: https://stackoverflow.com/a/6791968/768633

**Comment by VishalNehra** [comment]
Maybe adding it in constructor instead for each instance would be better.
Nice catch! I couldn't find the problem after spending much time.

**Comment by TranceLove** [comment]
But a question before going to conclusion: why `UtilsHandler` never had such problem? We asked it to create records early enough?

Lazy init kills I suppose...

**Comment by VishalNehra** [comment]
Not sure... I've done this in multiple projects without fail.. not sure why this problem all of a sudden.

**Comment by EmmanuelMess** [comment]
The call to `onUpgrade()` is not performed unless `getWritableDatabase();` is called, happed to me in SimpleAccounting. @ningshu please explain the line with a comment so that we don't delete it later.

**Comment by VishalNehra** [comment]
@EmmanuelMess please add commits to this PR, I think we can push this with v3.3.0 so let's not wait over these things.

**Comment by ningshu** [comment]
The problem of missing ‘sort’ table has been fixed.

**Comment by TranceLove** [comment]
Two ticks, merging.

**Pull request merged**